### PR TITLE
Improve Examples and Provider documentation

### PR DESCRIPTION
**Pull request by Dschee**
This **improves the documentation** of the Examples.rd and Provider.rd files by adding **more details**, structuring by params (**clarity**) and being more explicit about Moyas **default behavior** and **minimum usage requirements**.

I did these changes after trying to use Moya for a project and got stuck with it after reading the documentation alone. Once I looked into the code all my questions got answered quickly – but not everybody will do that. So I in this PR I try to be more explicit about the parts that confused me and try to provide **more detailed** and **practical examples**.

See also #432.

### CODE REVIEW

File: docs/Examples.md [hunk 1]
```diff
@@ -3,72 +3,100 @@ Examples
...
         }
     }
     var method: Moya.Method {
-        // all requests in this example will use GET.  Usually you would switch
-        // on the enum, like we did in `var path: String`
-        return .GET
+        switch self {
+        case .Zen, .ShowUser(_):
+            return .GET
+        case .UpdateUser(_, _, _):
```
Comment (ashfurrow) [inline docs/Examples.md]: If you don't need an enum's associated values, you can also just use `case .UpdateUser:` and it should work :+1:

File: docs/Examples.md [hunk 2]
```diff
@@ -3,72 +3,100 @@ Examples
...
+            return .PATCH
```
Comment (ashfurrow) [inline docs/Examples.md]: Patch eh? Nice, one of my favourite HTTP obscurities. For the sake of keeping the code familiar, do you think we should use a more typical put instead?

File: docs/Providers.md [hunk 3]
```diff
@@ -123,17 +133,21 @@ let manager = Manager(
...
 ```
 
+### plugins:
+
 Finally, you may also provide an array of `plugins` to the provider. These receive callbacks
 before a request is sent and after a response is received. There are a few plugins
 included already: one for network activity (`NetworkActivityPlugin`), one for logging
 all network activity (`NetworkLoggerPlugin`), and another for [HTTP Authentication](Authentication.md).
 
+For example you can enable the logger plugin by simply passing `[NetworkLoggerPlugin()]` alongside the `plugins` parameter of your `Endpoint`. Note that a plugin can also be configurable, for example the already included `NetworkActivityPlugin` requires a `networkActivityClosure` parameter. The configurable plugin implementation looks like this:
```
Comment (ashfurrow) [inline docs/Providers.md]: Could be worth linking to the [auth docs](https://github.com/Moya/Moya/blob/master/docs/Authentication.md#basic-http-auth) for further reading on the CredentialsPlugin.

**Comment by ashfurrow** [comment]
This is fantastic, @Dschee, thank you for taking the time to improve our documentation! I've made a few inline suggestions above – I'm happy to merge as-is and submit a PR with my feedback, none of these are blockers.

File: docs/Examples.md (hunk 1 above)
Comment (Dschee) [inline docs/Examples.md]: Cool, you're right, I was using it like that already but thought that might be a Swift side effect and actually undocumented. But now you say it, I can clean it up. Thanks!

File: docs/Examples.md (hunk 2 above)
Comment (Dschee) [inline docs/Examples.md]: Sure, but let's not use PUT either then as it's not very RESTful (which I think is kind of the standard right now) so what about changing the `.UpdateUser` case up to a `.CreateUser` and make it a POST? That should be familiar enough. I've changed the code to do this.

File: docs/Providers.md (hunk 3 above)
Comment (Dschee) [inline docs/Providers.md]: Isn't the `Authentication.md` linked just at the end of the section above? Shouldn't this be sufficient? You can of course add an additional link but I'm not sure what you are thinking of exactly.

**Comment by Dschee** [comment]
I've answered all your inline comments and tried to update the code accordingly. I hope all issues are fixed now. Thanks for considering this PR, @ashfurrow! :tada:

File: docs/Examples.md [hunk 4]
```diff
@@ -3,72 +3,100 @@ Examples
 
 So how do you use this library? Well, it's pretty easy. Just follow this
 template. First, set up an `enum` with all of your API targets. Note that you
-can include information as part of your enum. Let's look at a simple example.
+can include information as part of your enum. Let's look at a common example. First we create a new file named `MyService.swift`:
 
 ```swift
-enum GitHub {
+enum MyService {
```
Comment (orta) [inline docs/Examples.md]: :+1:

File: docs/Providers.md (hunk 3 above)
Comment (ashfurrow) [inline docs/Providers.md]: Ah, yes! Must've missed it.

**Comment by ashfurrow** [comment]
Thanks a lot for contributing to Moya! I've invited you to join the Moya GitHub organization – no pressure to accept! If you'd like more information on what that means, check out our [contributor guidelines](https://github.com/Moya/contributors) and feel free to reach out with any questions.

**Pull request merged**